virtual_ecosystem/models/plants/plants_model.py

```diff
@@ -651,7 +666,38 @@ def calculate_turnover(self) -> None:
         self.data["root_turnover_c_p_ratio"] = xr.full_like(
             self.data["elevation"], 656.7
         )
-
         self.data["nitrogen_fixation_carbon_supply"] = xr.full_like(
             self.data["elevation"], 0.01
         )
+        self.data["root_carbohydrate_exudation"] = xr.full_like(
+            self.data["elevation"], 0.025
+        )
+
+    def calculate_nutrient_uptake(self) -> None:
+        """Calculate uptake of soil nutrients by the plant community.
+
+        This function calculates the rate a which plants take up inorganic nutrients
+        (ammonium, nitrate, and labile phosphorus) from the soil.
+
+        Warning:
+            At present, this function just calculates uptake based on an entirely made
+            up function, and does not link to plant dynamics in any way.
+        """
+
+        # Assume plants can take 0.1% of the available nutrient per day, uptake stops
+        # (rather than goes negative) if the soil nutrient concentrations go negative
+        self.data["plant_ammonium_uptake"] = xr.where(
+            self.data["soil_n_pool_ammonium"] > 0.0,
+            self.data["soil_n_pool_ammonium"] * 0.001,
+            0.0,
```

Comment:
Should this be `self.data["soil_n_pool_ammonium"]` instead of zero? In this case, the plant exhaust the last amount of nutrient rather than not taking anything.

Reply:
This maybe isn't clear enough, but the `where` condition here is defining what happens when nutrient concentrations are negative, so this acts to stop plants from taking up a negative amount of nutrients (i.e. for some reason deciding to pump nutrients into the soil). This is basically just making sure that the nonsensical things don't happen when the "negative nutrient concentration" edge case is hit